packages/http-client-java/emitter/src/code-model-builder.ts

```diff
@@ -2753,14 +2761,31 @@ export class CodeModelBuilder {
       schema = this.processSchema(nonNullType, "");
     }
 
-    return new Property(prop.name, prop.doc ?? "", schema, {
+    const property = new Property(prop.name, prop.doc ?? "", schema, {
       summary: prop.summary,
       required: !prop.optional,
       nullable: nullable,
       readOnly: this.isReadOnly(prop),
       serializedName: prop.kind === "property" ? getPropertySerializedName(prop) : undefined,
       extensions: extensions,
     });
+
+    // xml
+    if (prop.kind === "property" && prop.serializationOptions.xml) {
+      // property.serializedName is set via getPropertySerializedName
+
+      // "serialization" is set to the property in TypeSpec emitter, not in the schema
+      // this avoid duplicate schema, when different property has different serialization options, but refers to the same schema
+      const propertyWithSerialization = property as Property & {
+        serialization?: {
+          xml?: XmlSerializationFormat;
+        };
+      };
+      propertyWithSerialization.serialization = propertyWithSerialization.serialization ?? {};
+      propertyWithSerialization.serialization.xml = getXmlSerializationFormat(prop);
```

Comment:
We do differently on code-model.

Here we set the serialization on property to `Property`.

m4 would set them to `Schema` of the `Property`. This causes problem on TypeSpec, as the underlying `Schema` can be shared across properites, which can have different xml serialization option, e.g.
```
@doc("Contains fields of arrays of primitive types.")
model ModelWithSimpleArrays {
  colors: string[];
  counts: int32[];
}

model ModelWithRenamedArrays {
  @name("Colors") @unwrapped colors: string[];
  @name("Counts") counts: int32[];
}
```